ConsoleApp1/Program.cs

```diff
@@ -232,8 +146,199 @@ static void SearchRecords(string path)
 					Console.WriteLine($"{record.firstName,-15} {record.lastName,-15} {record.phone,-15}");
 				}
 			}
+		}
+
+		static void DeleteRecords(string path)
+		{
+			DisplayPhoneBook(path);
+
+			Console.WriteLine();
+			Console.WriteLine("Deleting records");
+			Console.Write("Enter record index (0 to cancel): ");
 
-			//DisplayPhoneBook(searchedRecords);
+			int index;
+
+			if (int.TryParse(Console.ReadLine() ?? "1", out index))
+			{
+				if (index > 0)
+				{
+					(string firstName, string lastName, string phone)[] phoneBookRecords = ReadPhoneBookRecords(path);
+
+                    if (index - 1 < phoneBookRecords.Length)
+					{
+                        (string firstName, string lastName, string phone)[] res = 
+							new (string firstName, string lastName, string phone)[phoneBookRecords.Length - 1];
+
+                        for (int i = 0; i < phoneBookRecords.Length; i++)
+						{
+							if (i < index - 1)
+							{
+								res[i] = phoneBookRecords[i];
+
+							} else if (i > index - 1)
+							{
+								res[i - 1] = phoneBookRecords[i];
+							}
+						}
+
+						string[] strRes = new string[res.Length  + 1];
+						strRes[0] = "FirstName,LastName,PhoneNumber";
+
+                        for (int i = 0; i < res.Length; i++)
+						{
+							strRes[i + 1] = $"{res[i].firstName},{res[i].lastName},{res[i].phone}";
+						}
+                        File.WriteAllLines(path, strRes);
+					}
+				}
+
+            } else
+			{
+				Console.WriteLine("invalid input");
+			}
+        }
+
+		static void SortRecords(string path)
+		{
+
+            Func<(string firstName, string lastName, string phone), int, string> SortBy = 
+				(a, b) => b == 1 ? a.firstName : b == 2 ? 
+					a.lastName : a.phone;
+			Func<string, string, int, bool> sorter = (a, b, o) => o == 1 ?
```

Comment:
You should simplify your code as possible
Func<string, string, int, bool> sorter = (a, b, o) => string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 && o == 1;